source/material_model/grain_size.cc

```diff
@@ -852,8 +852,10 @@ namespace aspect
 
               if (HeatingModel::ShearHeatingOutputs<dim> *shear_heating_out = out.template get_additional_output<HeatingModel::ShearHeatingOutputs<dim>>())
                 {
-                  if (grain_size_evolution_formulation == Formulation::paleowattmeter ||
-                      grain_size_evolution_formulation == Formulation::paleopiezometer)
+                  AssertThrow(grain_size_evolution_formulation == Formulation::paleopiezometer,
+                              ExcMessage("Shear heating output is not created with the Paleopiezometer grain damage formulation."));
```

Comment:
Shouldnt this be:
```suggestion
                  AssertThrow(grain_size_evolution_formulation != Formulation::paleopiezometer,
                              ExcMessage("Shear heating output should not be used with the Paleopiezometer grain damage formulation."));
```
But it also looks like we will always create shear heating outputs (even if no shear heating is active) so this assert will always trigger.